src/Orleans.Transactions/DistributedTM/IClock.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+namespace Orleans.Transactions
+{
+    /// <summary>
+    /// System clock abstraction
+    /// </summary>
+    public interface IClock
```

Comment:
if `IClock` is introduced just for testing purpose , suggesting make it internal, and make the test project do internal visible to. If not, I'd like to understand the intention of the new public interface

Reply:
Testability is a design goal.  Having a clock abstraction as part of the public surface of the implementation violates no design principles I am aware of, and improves testability.  This is a common problem for components with dependencies on time.  There is not (to my knowledge) a common pattern for addressing this, but many systems do use clock abstractions for testability.

Reply:
hmm having abstraction is fine. Having a public abstraction just for testing isn't common to my understanding. If it is just for testing, can we keep it as a internal interface? this is what I meant by my previous comment.